compaction/compaction.cc

```diff
@@ -1900,4 +1901,16 @@ uint64_t compaction_descriptor::sstables_size() const {
     return boost::accumulate(sstables | boost::adaptors::transformed(std::mem_fn(&sstables::sstable::data_size)), uint64_t(0));
 }
 
+void compaction_descriptor::retrieve_owned_ranges_if_required(::compaction::compaction_state& cs) {
+    for (const auto& sst : sstables) {
+        if (sst->requires_cleanup()) {
+            if (!cs.owned_ranges_ptr) {
```

Comment:
yes, we could only check cs.sstables_requiring_cleanup. sstables and compaction_state as two sources for knowing the cleanup state is fragile.